scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.1.6
+#--
+
+DriveId1_6 := utils.PolicyIdWithSuffix("GWS.DRIVEDOCS.1.6")
+
+LogMessage1_6 := "SHARING_ACCESS_CHECKER_OPTIONS"
+
+Check1_6_OK if {
+    not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage1_6, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check1_6_OK if {PolicyApiInUse}
+
+NonComplianceMessage1_6(value) := sprintf("Access Checker allows users to share files to %s",
+                                          [value])
+
+GetFriendlyValue1_6(Value) :=
+"recipients only, suggested target audience, or public (no Google account required)" if {
+    Value in {"ALL", "RECIPIENTS_OR_AUDIENCE_OR_PUBLIC"}
+} else := "recipients only, or suggested target audience" if {
+    Value in {"DOMAIN_OR_NAMED_PARTIES", "RECIPIENTS_OR_AUDIENCE"}
+} else := Value
+
+NonCompliantOUs1_6 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_6(GetFriendlyValue1_6(LastEvent.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage1_6, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    AcceptableValues := {"NAMED_PARTIES_ONLY", "INHERIT_FROM_PARENT"}
+    not LastEvent.NewValue in AcceptableValues
+}
+
+NonCompliantGroups1_6 contains {
+    "Name":Group,
+    "Value": NonComplianceMessage1_6(GetFriendlyValue1_6(LastEvent.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage1_6, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    AcceptableValues := {"NAMED_PARTIES_ONLY", "INHERIT_FROM_PARENT"}
+    not LastEvent.NewValue in AcceptableValues
+}
+
+NonCompliantOUs1_6 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_6(GetFriendlyValue1_6(accessCheck))
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    accessCheck := settings.drive_and_docs_external_sharing.accessCheckerSuggestions
+    accessCheck != "RECIPIENTS_ONLY"
+}
+
+tests contains {
+    "PolicyId": DriveId1_6,
+    "Criticality": "Shall",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := false
+    not Check1_6_OK
+}
+
+tests contains {
+    "PolicyId": DriveId1_6,
+    "Criticality": "Shall",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_6, NonCompliantGroups1_6),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs1_6,
+        "NonCompliantGroups": NonCompliantGroups1_6},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check1_6_OK
+    Conditions := {count(NonCompliantOUs1_6) == 0, count(NonCompliantGroups1_6) == 0}
+    Status := (false in Conditions) == false
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.1.7
+#--
```

Comment:
If you have sharing set to OFF but had "Distributing content outside of..." enabled first, 1.7 still fails.

Reply:
done